src/engraving/rw/compat/compatutils.cpp

```diff
@@ -235,3 +238,47 @@ void CompatUtils::replaceOldWithNewOrnaments(MasterScore* score)
         delete oldOrnament;
     }
 }
+
+void CompatUtils::replaceOldWithNewExpressions(MasterScore* score)
+{
+    std::vector<StaffText*> oldExpressions; // Expressions used to be staff text
+
+    for (Measure* measure = score->firstMeasure(); measure; measure = measure->nextMeasure()) {
+        for (Segment& seg : measure->segments()) {
+            for (EngravingItem* item : seg.annotations()) {
+                if (item && item->isStaffText() && toStaffText(item)->textStyleType() == TextStyleType::EXPRESSION) {
+                    oldExpressions.push_back(toStaffText(item));
+                    LinkedObjects* links = item->links();
+                    if (!links || links->empty()) {
+                        continue;
+                    }
+                    for (EngravingObject* linkedItem : *links) {
+                        if (linkedItem != item) {
+                            oldExpressions.push_back(toStaffText(linkedItem));
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    for (StaffText* oldExpression : oldExpressions) {
+        Segment* parentSegment = toSegment(oldExpression->parentItem(true));
+
+        Expression* newExpression = Factory::createExpression(score->dummy()->segment());
+        newExpression->setParent(parentSegment);
+        newExpression->setTrack(oldExpression->track());
+        newExpression->setXmlText(oldExpression->xmlText());
+        newExpression->mapPropertiesFromOldExpressions(oldExpression);
+
+        LinkedObjects* links = oldExpression->links();
+        newExpression->setLinks(links);
+        if (links) {
+            links->push_back(newExpression);
+        }
```

Comment:
Yeah it feels somewhat better to me to replace each single object and hook it up to the existing `links`, rather than making new clones from the "main" object